src/base/memory_monitor_sr.cpp

```diff
@@ -402,30 +472,110 @@ namespace cubperf
     m_mmon = mmon;
   }
 
-  int memory_monitor::aggregater::get_server_info (const MMON_SERVER_INFO &info)
+  void memory_monitor::aggregater::get_server_info (MMON_SERVER_INFO &info) const
   {
-    return 0;
+    strncpy (info.name, m_mmon->m_server_name.c_str (), m_mmon->m_server_name.size ());
+    info.total_mem_usage = m_mmon->m_total_mem_usage.load ();
   }
 
-  int memory_monitor::aggregater::get_module_info (const MMON_MODULE_INFO &info, int module_index)
+  int memory_monitor::aggregater::get_module_info (MMON_MODULE_INFO &info, int module_index) const
   {
-    return 0;
+    int error = NO_ERROR;
+
+    error = m_mmon->m_module[module_index]->aggregate_stats (info, false);
+    return error;
   }
 
-  int memory_monitor::aggregater::get_transaction_info (const MMON_TRAN_INFO &info, int tran_count)
+  int memory_monitor::aggregater::get_module_brief_info (MMON_MODULE_INFO *&info, int option) const
   {
-    return 0;
+    int error = NO_ERROR;
+
+    for (int i = 0; i < MMON_MODULE_LAST; i++)
+      {
+	error = m_mmon->m_module[i]->aggregate_stats (info[i], true);
+	if (error != NO_ERROR)
+	  {
+	    return error;
+	  }
+      }
+
+    if (MMON_MODULE_DEFAULT_OPTION)
```

Comment:
1. What is the default?
2. The default could be changed. Let's use more specific conditions in the code.